hazelcast-sql/src/main/java/com/hazelcast/sql/impl/schema/type/Type.java

```diff
@@ -139,7 +138,7 @@ public void readData(ObjectDataInput in) throws IOException {
             String typeName = in.readString();
             type = converter.getTypeFamily() == QueryDataTypeFamily.OBJECT && typeName != null
                     ? new QueryDataType(typeName)
-                    : QueryDataTypeUtils.resolveTypeForClass(converter.getValueClass());
+                    : QueryDataType.resolveForConverter(converter);
```

Comment:
not caused by this PR but do we need to write `typeName` for types other than `OBJECT`?

Reply:
This will be improved in HZ-3944.